### PR TITLE
Log improvements

### PR DESCRIPTION
**Pull request by raych1**
- cleaned up handlebar templates
- added a function to save the filtered log
- introduced `loggerWaitToFinish` to ensure log transports complete
- included version in the `SdkAutoOptions`
- enhanced logging details in report status generation
- resolved conflicts in the CLI alias of `v`
- removed unused `CommentCaptureTransport`

### CODE REVIEW

File: tools/spec-gen-sdk/src/automation/reportStatus.ts [hunk 1]
```diff
@@ -92,9 +84,62 @@ export const generateReport = (context: WorkflowContext) => {
   }
 
   context.logger.log('endsection', 'Generate report');
-  //context.logger.remove(captureTransport);
 }
 
+export const saveFilterLog = async (context: WorkflowContext) => {
```
Comment (chidozieononiwu) [inline tools/spec-gen-sdk/src/automation/reportStatus.ts:89]: Should this be saveFilteredLog?

File: tools/spec-gen-sdk/src/cli/cli.ts [hunk 2]
```diff
@@ -173,7 +176,7 @@ yargs(hideBin(process.argv))
           description: "The branch of the head repository of the specification pull request",
         },
         'api-version': {
-          alias: "v",
+          alias: "apiv",
```
Comment (chidozieononiwu) [inline tools/spec-gen-sdk/src/cli/cli.ts:179]: You could just make it av

**Review by chidozieononiwu — APPROVED** [review]
Few Suggestions otherwise it looks fine to me

File: tools/spec-gen-sdk/src/automation/reportStatus.ts (hunk 1 above)
Comment (raych1) [reply tools/spec-gen-sdk/src/automation/reportStatus.ts]: Updated it and renamed others.

File: tools/spec-gen-sdk/src/cli/cli.ts (hunk 2 above)
Comment (raych1) [reply tools/spec-gen-sdk/src/cli/cli.ts:179]: I intended not using 'av' for potential Policheck alert.

**Pull request merged**